File: backend/src/models/UserArticleInteraction.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export interface IUserArticleInteraction extends Document {
    userId: number;
    articleId: Types.ObjectId;
    isLiked: boolean;
    isCollected: boolean;
}

const UserArticleInteractionSchema: Schema = new Schema({
    userId: {
        type: Number,
        required: [true, '用户ID不能为空'],
        trim: true
    },
    articleId: {
        type: Types.ObjectId,
        required: [true, '文章ID不能为空'],
        trim: true
    },
    isLiked: {
        type: Boolean,
        default: true
    },
    isCollected: {
        type: Boolean,
        default: true
    }
});

const UserArticleInteraction = mongoose.model<IUserArticleInteraction>('user_article_interactions', UserArticleInteractionSchema);
export default UserArticleInteraction;

File: backend/java-backend/src/main/java/com/example/demo/model/mongo/UserArticleInteraction.java
package com.example.demo.model.mongo;

import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.CompoundIndex;

@Data
@Document(collection = "user_article_interactions")
@CompoundIndex(name = "userId_articleId_unique", def = "{'userId': 1, 'articleId': 1}", unique = true)
public class UserArticleInteraction {

    @Id
    private String id;

    private Integer userId;
    // articleId 字段类型为 ObjectId，与 Article._id 类型保持一致，避免 MongoDB 类型不匹配
    private ObjectId articleId;
    private Boolean isLiked;
    private Boolean isCollected;
}
